tools/python/clean_strings_txt.py

```diff
@@ -0,0 +1,329 @@
+from __future__ import print_function
+
+import logging
+import re
+import subprocess
+from argparse import ArgumentParser
+from collections import defaultdict
+from itertools import chain
+from os.path import abspath, isabs
+
+from find_untranslated_strings import StringsTxt
+
+"""
+This script determines which strings are used in the platform code (iOS and
+Android) and removes all the other strings from strings.txt. For more information,
+run this script with the -h option.
+"""
+
+
+OMIM_ROOT = ""
+
+MACRO_RE =  re.compile('L\(.*?@\"(.*?)\"\)')
+XML_RE = re.compile("value=\"(.*?)\"")
+ANDROID_JAVA_RE = re.compile("R\.string\.([\w_]*)")
+ANDROID_XML_RE = re.compile("@string/(.*?)\W")
+IOS_CANDIDATES_RE = re.compile("(.*?):[^L\(]@\"([a-z0-9_]*?)\"")
+
+HARDCODED_CATEGORIES = None
+
+HARDCODED_COLOURS = [
+    "red", "yellow", "blue", "green", "purple", "orange", "brown", "pink"
+]
+
+
+def exec_shell(test, flags):
+    spell = ["{0} {1}".format(test, flags)]
+
+    process = subprocess.Popen(
+        spell,
+        stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+        shell=True
+    )
+
+    logging.info(" ".join(spell))
+    out, _ = process.communicate()
+    return filter(None, out.splitlines())
+
+
+def grep_ios():
+    logging.info("Grepping ios")
+    grep = "grep -r 'L(\|localizedText\|localizedPlaceholder' {0}/iphone/*".format(OMIM_ROOT)
+    ret = exec_shell(grep, "")
+    return filter_ios_grep(ret)
+
+
+def grep_android():
+    logging.info("Grepping android")
+    grep = "grep -r 'R.string.' {0}/android/src".format(OMIM_ROOT)
+    ret = android_grep_wrapper(grep, ANDROID_JAVA_RE)
+    grep = "grep -r '@string/' {0}/android/res".format(OMIM_ROOT)
+    ret.update(android_grep_wrapper(grep, ANDROID_XML_RE))
+    grep = "grep -r '@string/' {0}/android/AndroidManifest.xml".format(OMIM_ROOT)
+    ret.update(android_grep_wrapper(grep, ANDROID_XML_RE))
+
+    return parenthesize(ret)
+
+
+def grep_ios_candidates():
+    logging.info("Grepping ios candidates")
+    grep = "grep -nr '@\"' {0}/iphone/*".format(OMIM_ROOT)
+    ret = exec_shell(grep, "")
+
+    strs = strings_from_grepped(ret, IOS_CANDIDATES_RE)
+    return strs
+
+
+def android_grep_wrapper(grep, regex):
+    grepped = exec_shell(grep, "")
+    return strings_from_grepped(grepped, regex)
+
+
+def filter_ios_grep(strings):
+    with_no_blobs = filter(lambda s: not s.startswith("Binary file"), strings)
+    filtered = strings_from_grepped(with_no_blobs, MACRO_RE)
+    filtered = parenthesize(process_ternary_operators(filtered))
+    filtered.update(parenthesize(strings_from_grepped(with_no_blobs, XML_RE)))
+    filtered.update(parenthesize(HARDCODED_CATEGORIES))
+    filtered.update(parenthesize(HARDCODED_COLOURS))
```

Comment:
They are represented differently in strings.txt and in the client code: just the name of the colour in the strings.txt file and "color_smth" on the client. This makes it impossible to find them like other strings.